test/e2e_node/oomkiller_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+type testCase struct {
+	podSpec                *v1.Pod
+	oomTargetContainerName string
+}
+
+const PodOOMKilledTimeout = 2 * time.Minute
+
+var _ = SIGDescribe("OOMKiller [LinuxOnly] [NodeConformance]", func() {
```

Comment:
We likely want to make this [Disruptive] as well.

Reply:
OOM killer would always kill this specific container as when it exceeds limits. OOM killer may also kill another container when the system is running low on memory, but this theoretically can be triggered by any container, not only this one. WDYT?

Reply:
it's a single container failure. No disruption for other tests.

Reply:
+1, this is not an issue since it is a container cgroup OOM.